artiq/devices/novatech409B/novatech409B.py

```diff
@@ -0,0 +1,362 @@
+
+        # setup logging
+        FORMAT = "%(asctime)-15s %(message)s"
+        logging.basicConfig(format=FORMAT)
+        self.logger = logging.getLogger("artiq.driver.novatech409B")
+        self.logger.setLevel(logging.DEBUG)
+        self.debug_message("__init__", "", level=3)
+
+        # establish serial connection --- platform dependent
+        if not self.simulate_hw:
+            self.__platform = platform.system()
+            if self.__platform == "Windows":
+                # note that pySerial starts counting serial ports at 0
+                serial_port_id = int(self.__comport)-1
+            elif self.__platform is "Linux":
+                # just use the device string as
+                # passed (e.g. "/dev/ttyUSB0")
+                serial_port_id = "/dev/ttyUSB0"
+            else:
+                self.debug_message("__init__", "unknown platform", level=0)
+                sys.exit()
+            self.__ser = serial.Serial(
+                serial_port_id,
+                baudrate=19200,
+                bytesize=8,
+                parity="N",
+                stopbits=1,
+                xonxoff=0,
+                timeout=0.05)
+        self.setup()
+
+    def __del__(self):
+        if not self.simulate_hw:
+            self.__ser.close()
+            time.sleep(1)
+
+    def echo(self, s):
+        ss = "novatech409B.echo() :: " + s
+        self.debug_message("echo", ss)
+        return ss
+
+    def debug_message(self, func_name, msg, level=2):
+        """generate debug message
+
+        :param str func_name: the calling function"s name
+        :param str msg: is a message
+        :param int level: is the debug level
+            ** 2 information helpful in typical use scenario
+            ** 3 full-on debug info (annoying)
+        :returns: None
+        """
+        if 1:  # (level <= self.__debug) :
+            # try a trick to automatically infer caller name
+            inferred_func_name = inspect.stack()[2][3]
+            s = inferred_func_name + "() :: " + msg
+            # here"s what a typical warnning message looks like:
+            # WARNING:ARTIQ.driver.novatech409B:
+            #      set_phase_continuous() :: M n
+            self.logger.warning(s)
+
+    def ser_send(self, myStr, ignore_unusual_response=False):
+        """send a string to the serial port
+
+        Routine for sending serial commands to device. It sends strings
+        and listens for a response terminated by a carriage return.
+
+        example:
+        ser_send("F0 1.0") #sets the freq of channel 0 to 1.0 MHz
+
+        :param str myStr: a character string to send to device
+        :returns: None
+        """
+        self.debug_message("ser_send", myStr, level=3)
+        s = myStr + "\r\n"
+        expected_response = b"OK\r\n"
+        result = b""
+        if self.simulate_hw is False:
+            try:
+                self.__ser.flush()
```

Comment:
why? i doubt this is useful or needed. would you not want to flush after `write()`?